### PR TITLE
Remove redundant non-trivial Variant types initializations

### PR DESCRIPTION
**Pull request by Sauermann**
As mentioned in https://github.com/godotengine/godot/pull/68627#discussion_r1021739394.

### CODE REVIEW

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
While at it, there's also:
```
scene/resources/skeleton_modification_2d_fabrik.h
71:     Transform2D target_global_pose = Transform2D();
72:     Transform2D origin_global_pose = Transform2D();
```

And some more for other vector/matrix types:
```
$ rg "Transform3D .* = Transform3D\(\);" --sort path
core/variant/variant_call.cpp
2441:   Transform3D identity_transform = Transform3D();

drivers/gles3/rasterizer_scene_gles3.h
102:    Transform3D cam_transform = Transform3D();
103:    Transform3D inv_cam_transform = Transform3D();

editor/plugins/node_3d_editor_plugin.cpp
3495:           Transform3D axis_angle = Transform3D();

editor/plugins/skeleton_3d_editor_plugin.cpp
1291:   Transform3D original_to_local = Transform3D();
1299:   Transform3D t = Transform3D();

scene/resources/skeleton_modification_3d_fabrik.h
71:     Transform3D target_global_pose = Transform3D();
72:     Transform3D origin_global_pose = Transform3D();
```

And all other non-trivial Variant types such as `Vector*`, `String`, `Color`, `StringName`, etc. all have similar occurrences. But I guess that's out of scope for this PR :)

**Comment by Sauermann** [comment]
> And all other non-trivial Variant types such as Vector*, String, Color, StringName, etc. all have similar occurrences. But I guess that's out of scope for this PR :)

I can simply rename the PR to "Remove redundant initialization" to change the scope ;-)
Will see what I can find.

File: editor/plugins/animation_state_machine_editor.cpp [hunk 1]
```diff
@@ -420,7 +420,7 @@ void AnimationNodeStateMachineEditor::_state_machine_gui_input(const Ref<InputEv
 	if (mm.is_valid()) {
 		state_machine_draw->grab_focus();
 
-		String new_over_node = StringName();
+		String new_over_node;
```
Comment (Sauermann) [inline editor/plugins/animation_state_machine_editor.cpp:423]: Initializing a `String` with `StringName()` was probably not intentional.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**